### PR TITLE
Added allowed_workflows to pytorch probot

### PR DESCRIPTION
**Pull request by DanilBaibak**
Added allowed_workflows to pytorch probot. This is a follow up PR [regarding the retry bot](https://github.com/pytorch/test-infra/pull/3942/files#diff-ee5e4f1e1fa962c6f62e5dcebde6e0bab573e74474601bf5749ccb668fd9c900R14-R16).

### CODE REVIEW

File: .github/pytorch-probot.yml [hunk 1]
```diff
@@ -1 +1,8 @@
 tracking_issue: 2447
+allowed_workflows:
+- lint
```
[the remark this answers is not in the document]
Comment (DanilBaibak) [reply .github/pytorch-probot.yml]: It is not case sensitive. I removed the lint workflow from the list. Once the configuration is moved to the domain repo, you guys decide which worker processes should be restarted.

File: .github/pytorch-probot.yml [hunk 2]
```diff
@@ -1 +1,10 @@
 tracking_issue: 2447
+
+# List of workflows that will be re-run in case of failures
+# https://github.com/pytorch/test-infra/blob/main/torchci/lib/bot/retryBot.ts
+allowed_workflows:
+- Build Linux
+- Build Macos
+- Build M1
+- Tests on Linux
+- Tests on macOS
```
[the remark this answers is not in the document]
Comment (DanilBaibak) [reply .github/pytorch-probot.yml:10]: I just prefer to be a bit more explicit. But again, this config should be maintained by domains. For now, I need it to be present. After, you can change it according to your needs or requirements.

**Review by pmeier — APPROVED** [review]
To summarize, all workflows going by the `name: "..."` attribute listed here, will be retried on the `main` branch in case they fail, right? Meaning, if we have flaky tests, the `viable/strict` branch can be updated anyway, correct?

**Comment by DanilBaibak** [comment]
No, the retry mechanism should work for all PRs. But there are some additional rules: the number of failed jobs should be less than 5 per workflow, retry can happen once for workflow.

**Pull request merged**